includes/gateway/class-omise-payment-fbbot.php

```diff
@@ -0,0 +1,266 @@
+		      	$image_url = site_url() . '/wp-content/plugins/omise-woocommerce/assets/images/omise_logo.png';
+		      	$url = 'https://www.messenger.com/closeWindow/?image_url=' . $image_url . '&display_text=THANKS%20FOR%20PURCHASE';
+		      	if ( wp_redirect( $url ) ) {
+		        	exit;
+		      	}
+
+		      	return $posts;
+		    }
+
+		    // Create custom page
+		    $post = new stdClass;
+
+		    if ( $this->is_omise_payment_page() ) {
+		    	$post->post_title = __( 'Your order' );
+		    	$post->post_content = $this->payment_page_render();
+
+		    } else if ( $this->is_omise_payment_error_page() ) {
+		      	$post->post_title = __( 'System error' );
+		      	$post->post_content = $this->payment_error_page_render();
+		    }
+
+		    if ( $this->is_accessible() ) {
+				// Create cumtom page content
+				$post->post_author = 1;
+				$post->post_name = strtolower( $wp->request );
+				$post->guid = get_bloginfo( 'wpurl' ) . '/' . strtolower( $wp->request );
+				$post->ID = -1;
+				$post->post_type = 'page';
+				$post->post_status = 'static';
+				$post->comment_status = 'closed';
+				$post->ping_status = 'open';
+				$post->comment_count = 0;
+				$post->post_date = current_time( 'mysql' );
+				$post->post_date_gmt = current_time( 'mysql', 1 );
+				$posts = array();
+				$posts[] = $post;
+
+				// make wp_query
+				$wp_query->is_page = true;
+				$wp_query->is_singular = true;
+				$wp_query->is_home = false;
+				$wp_query->is_archive = false;
+				$wp_query->is_category = false;
+				unset( $wp_query->query["error"] );
+				$wp_query->query_vars["error"] = "";
+				$wp_query->is_404 = false;
+		    }
+
+	    	return $posts;
+		}
+
+		public function payment_page_render () {
+			global $wp_query;
+
+			if ( ! isset( $wp_query->query_vars['product_id'] ) || ! isset( $wp_query->query_vars['messenger_id'] ) ) {
+				return '<strong>404 Your order not found</strong>';
+			}
+
+			$product_id = $wp_query->query_vars['product_id'];
+			$messenger_id = $wp_query->query_vars['messenger_id'];
+
+			$url = plugin_dir_path( dirname( __DIR__ ) ) . 'templates/fbbot/payment-form.php';
+			ob_start();
+			include( $url );
+			return ob_get_clean();
+		}
+
+		public function payment_error_page_render () {
+			global $wp_query;
+
+			if ( ! isset( $wp_query->query_vars['error_message'] ) ) {
+			  return '<strong>Woocommerce system has error. Please try again.</strong>';
+			}
+
+			$error_message = $wp_query->query_vars['error_message'];
+
+			return '<strong>' . $error_message . '</strong>';
+		}
+
+		public function process_payment_by_bot( $params ) {
+		    /* rearrange step
+		      1. receive post data from payment page
+		      2. Create wc order
+		      3. Create charge 
+		      4. update order status
+		    */
+
+		    $omise_token = $params['omise_token'];
+		    $product_id = $params['product_id'];
+		    $messenger_id = $params['messenger_id'];
+
+		    $order = Omise_FBBot_WooCommerce::create_order( $params );
```

Comment:
Good thing, just ask because we have 2 choice to let them know. 
And about for show error on the error page, it no need to use more step like sent message to user via messenger app. 👍